cli/src/tunnels/socket_signal.rs

```diff
@@ -241,23 +284,25 @@ where
 		}
 	}
 
-	pub fn process(&mut self, contents: &[u8]) -> std::io::Result<&[u8]> {
+	pub fn process(&mut self, contents: &[u8], finish: bool) -> std::io::Result<&[u8]> {
 		let mut out_offset = 0;
 		let mut in_offset = 0;
 		loop {
 			let in_before = self.flate.total_in();
 			let out_before = self.flate.total_out();
 
-			match self
-				.flate
-				.process(&contents[in_offset..], &mut self.output[out_offset..])
-			{
+			match self.flate.process(
+				&contents[in_offset..],
+				&mut self.output[out_offset..],
+				finish,
+			) {
 				Ok(flate2::Status::Ok | flate2::Status::BufError) => {
 					let processed_len = in_offset + (self.flate.total_in() - in_before) as usize;
 					let output_len = out_offset + (self.flate.total_out() - out_before) as usize;
-					if processed_len < contents.len() {
+					if processed_len < contents.len() || output_len == self.output.len() {
```

Comment:
This was the real issue, the signaling of finish to the compressors was just for cleanliness/issues that might not be visible